src/Features/LanguageServer/ProtocolUnitTests/SemanticTokens/AbstractSemanticTokensTests.cs

```diff
@@ -31,6 +34,12 @@ private static LSP.SemanticTokensRangeParams CreateSemanticTokensRangeParams(LSP
                 Range = range
             };
 
+        protected static async Task UpdateDocumentTextAsync(string updatedText, Workspace workspace)
+        {
+            var docId = ((TestWorkspace)workspace).Documents.First().Id;
+            await ((TestWorkspace)workspace).ChangeDocumentAsync(docId, SourceText.From(updatedText));
```

Comment:
What's with all the casting to a concrete type here? We can't pass this in as a TestWorkspace?

Reply:
bad merge, fixed